### ISSUE TITLE
The DICOM File metadata dialog freezes for several seconds when searching for a string on large metadata

### ISSUE THREAD

**Issue by jadh4v** [Slicer/Slicer#7370]
## Summary
The `DICOM File metadata` dialog freezes for several seconds when searching for a string. This happens for a dicom file that has a significantly large size of metadata. In my use-case, this was a segmentation modality multiframe dicom which has reference items to individual SOP instances (ReferencedSOPClassUID, ReferencedSOPInstanceUID) of the original series that it is referencing.

The dialog freezes for several seconds depending on how you start typing in the search box (from 5 to 15 seconds).
The dialog goes "Not Responding" during this time.
<img width="800" alt="image" src="https://github.com/Slicer/Slicer/assets/22624785/bffd66c8-ece0-4662-a960-e5d9aa261477">

## Steps to reproduce

1. Find a dicom file that has a large metadata footprint (can use [this one from TCIA](https://data.kitware.com/#user/61d87b684acac99f42cd366c/folder/654d0c3dfdc1508d8bec1baf))
2. Import the file into dcm database of slicer
3. Find the segmentation series, right-click and open metadata.
4. Type in the search box (preferably in 2 or 3 rapid separate strokes).
5. Expect the search as you type feature to freeze the entire dialog for a short time between 5 to 15 seconds.

## Environment
- Slicer version: Slicer-5.5.0-2023-11-08
- Operating system: Windows 11 Pro 23H2 build: 22631.2506

**Comment by jcfr** [comment]
> dialog freezes for several seconds 

The work currently done by @Punzo may help:
* https://github.com/commontk/CTK/pull/1142

**Comment by jcfr** [comment]
For additional context, I checked with @jadh4v and the `Quantitative Reporting` extension installed.

The associated plugin sources are available here: https://github.com/QIICR/QuantitativeReporting/blob/master/DICOMPlugins/DICOMSegmentationPlugin.py


For reference, here is the check suggestion user to install the extension:

https://github.com/Slicer/Slicer/blob/0203cff2af9b11323523bc639800e3a597bcc3e9/Modules/Scripted/DICOMPlugins/DICOMScalarVolumePlugin.py#L282-L283

**Comment by Punzo** [comment]
> > dialog freezes for several seconds
> 
> The work currently done by @Punzo may help:
> 
> * [Add dicom visual navigation commontk/CTK#1142](https://github.com/commontk/CTK/pull/1142)

yes all the operations (query, retrieve, etc..) will in background, so it will not freeze the UI. I am still working on the PR. Hopefully, we will be able to merge asap (:

**Comment by pieper** [comment]
I don't think this is related to the Quantitive Reporting Extension or actually even to the changes @Punzo is working on, since the metadata dialog operates directly on data from the dicom files on disk.  The metadata for some dicom instances, like SEG and some SR can be very large but the dialog's search function can take a very long time.

The metadata dialog was originally implemented to work with collections of single frame instances.but may need efficiency tweaks or even just a progress bar to work with these large instances.

**Comment by jadh4v** [comment]
@pieper Yea, I think the _search-as-you-type_ needs to run in a separate non-blocking thread, so that the user experience is better even if the search will take some time.

**Comment by Punzo** [comment]
> actually even to the changes @Punzo is working on, since the metadata dialog operates directly on data from the dicom files on disk.

oh you are right, sorry I did not read the full post.

**Comment by lassoan** [comment]
These SEG information objects contain an absurd amount of mostly useless metadata. This huge amount of metadata causes performance issues _everywhere_ and the proper fix is to revise the DICOM standard. The problem is already known by the DICOM standard working groups and I believe there is ongoing work to address it.

Since metadata viewer performance is good on most clinical data sets, and it is mostly just a debugging tool anyway (it should not be necessary to look at the DICOM tags ever if everything works well), I don't think it is worth investing much time into its optimization.

Of course, if someone has time to do profiling and fix some performance bottlenecks without making the design much more complex then we would be happy to receive the pull request.
